core/templates/dev/head/services/questions-list.service.ts

```diff
@@ -81,6 +81,16 @@ angular.module('oppia').factory('QuestionsListService', [
           QuestionBackendApiService.fetchQuestionSummaries(
             skillIds, _nextCursorForQuestions).then(
             function(returnObject) {
+              if (skillIds.length === 1) {
+                for (var idx in returnObject.questionSummaries) {
+                  returnObject.questionSummaries[idx].skillDifficulty =
```

Comment:
Er, we shouldn't be modifying an object midstream. Better to have fetchQuestionSummary() and fetchQuestionSummaries() and have each of them return the appropriate thing (either a domain object, or a list of domain objects).

Reply:
`questionSummary` can be more than one independent of skillIds (1 skill linked to multiple questions). What did change was that now, only one linked skill is considered instead of all linked skills to the question. 
So, removed this part and moved this to the backend to not return single element arrays, but the elements themselves if length of `skillIds` is 1.

Reply:
Done, created `QuestionSummary` and `QuestionSummaryForOneSkill` domain objects.